packages/perseus/src/widgets/interactive-graphs/stateful-mafs-graph.tsx

```diff
@@ -31,6 +31,7 @@ export type StatefulMafsGraphProps = {
     showTooltips: Required<InteractiveGraphProps["showTooltips"]>;
     showProtractor: boolean;
     labels: InteractiveGraphProps["labels"];
+    hintMode: boolean;
```

Comment:
`InteractiveGraphProps["hintMode"]` is equivalent to `boolean`, as far as the TypeScript compiler is concerned. But the meaning of `boolean` is immediately obvious to anyone who knows TypeScript, while to find out what `InteractiveGraphProps["hintMode"]` means, you have to go digging through two other files. You also have to know that `hintMode` is common to all widgets and not a special interactive-graph prop; otherwise, you're unlikely to look in the right place. I dislike having to dig through layers of type declarations to figure out what the actual type of something is, so I prefer to write types like `boolean` instead of `InteractiveGraphProps["hintMode"]`.

Reply:
I always assumed it was done that way so that if the type of one changed it updated the type of the other. I think there are probably other ways of making sure to update it in all the right places though, such as tests and such, right?

Reply:
The type system / `tsc` itself will ensure that the types are kept consistent. That's the main benefit of a type system - keeping different things consistent 😄

Reply:
`InteractiveGraphProps` and all of the types defined in `perseus-types.ts` should not include props that the `Renderer` provides programmatically. The types in `perseus-types.ts` could be thought of as the data schema for Perseus (the serialized, at rest data format). 

So Ben has it the right way here. 

In the widget definitions, we've used the format `InteractiveGraphProps["some_prop"]` as a signal that the prop comes from the serialized widget options as opposed to something that is derived at runtime. All of these types/props could use a big cleanup/standardization, but it didn't get done during our Perseus Stabilization project and hasn't become a high enough priority yet. 

I'd love to see a "gold standard" widget definition that included a convention for defining all the necessary types, file layout, test file location, etc. :) #someday.